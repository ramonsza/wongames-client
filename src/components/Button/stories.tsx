import { Meta, StoryObj } from '@storybook/react'
import Button from '.'

export default {
  title: 'Button',
  component: Button
} as Meta

type Story = StoryObj<typeof Button>

export const Default: Story = {
  args: {
    children: 'Buy Now'
  }
}
