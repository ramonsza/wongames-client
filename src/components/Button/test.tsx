import { screen } from '@testing-library/react'
import { renderWithTheme } from '@/utils/tests/helpers'

import Button from '.'

describe('<Button />', () => {
  it('should render the medium size button by default', () => {
    const { container } = renderWithTheme(<Button>Buy Now</Button>)

    expect(screen.getByRole('button', { name: /Buy Now/i })).toHaveStyle({
      height: '4rem',
      padding: '0.8rem 3.2rem',
      'font-size': '1.4rem'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the small size', () => {
    const { container } = renderWithTheme(
      <Button $size="small">Buy Now</Button>
    )

    expect(screen.getByRole('button', { name: /Buy Now/i })).toHaveStyle({
      height: '3rem',
      'font-size': '1.2rem'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the large size', () => {
    const { container } = renderWithTheme(
      <Button $size="large">Buy Now</Button>
    )

    expect(screen.getByRole('button', { name: /Buy Now/i })).toHaveStyle({
      height: '5rem',
      padding: '0.8rem 4.8rem',
      'font-size': '1.6rem'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a full width version', () => {
    const { container } = renderWithTheme(<Button $fullWidth>Buy Now</Button>)

    expect(screen.getByRole('button', { name: /Buy Now/i })).toHaveStyle({
      width: '100%'
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
