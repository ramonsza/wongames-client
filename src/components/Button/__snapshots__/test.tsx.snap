// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the large size 1`] = `
.c0 {
  background: linear-gradient(180deg, #ff5f5f 0%, #f062c0 50%);
  color: #FAFAFA;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
}

<button
  class="c0"
>
  <span>
    Buy Now
  </span>
</button>
`;

exports[`<Button /> should render the medium size button by default 1`] = `
.c0 {
  background: linear-gradient(180deg, #ff5f5f 0%, #f062c0 50%);
  color: #FAFAFA;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

<button
  class="c0"
>
  <span>
    Buy Now
  </span>
</button>
`;

exports[`<Button /> should render the small size 1`] = `
.c0 {
  background: linear-gradient(180deg, #ff5f5f 0%, #f062c0 50%);
  color: #FAFAFA;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  height: 3rem;
  font-size: 1.2rem;
}

<button
  class="c0"
>
  <span>
    Buy Now
  </span>
</button>
`;
